### DISCUSSION TITLE
huber loss

### DISCUSSION THREAD

**Discussion by iamryanchia** [rwightman/efficientdet-pytorch#197] [Q&A] [answered]
The default box loss is huber loss. Through google, I know that the smooth l1 loss is also called huber loss. But the huber loss implemented in code does not look like smooth l1 loss, is this what you did intentionally? 

Have you trained with smooth l1 loss and what is the result? And I see `num_positives` multiplied by 4.0 as the `normalizer`. Generally, it will not be multiplied by 4.0. Does multiplying by 4.0 have a big impact on the result?

https://github.com/rwightman/efficientdet-pytorch/blob/b3dcb20e45a4d5e3ad35146a9723de2d41aae999/effdet/loss.py#L132

**Comment by rwightman** [comment, marked as answer]
huber and smooth l1 are similar but not quite, they are exactly the same when delta=beta=1.0, otherwise they differ. I had smooth l1 to try it at some diff beta but never ended up exploring that.

The box loss scaling values are as per the original tf retinanet impl that efficientdet was based on.

**Reply by iamryanchia** [reply]
I see, thank you very much!
